### ISSUE TITLE
Typo in Curb instrumentation causes ArgumentError

### ISSUE THREAD

**Issue by knarewski** [newrelic/newrelic-ruby-agent#1033]
## Description
When instrumented by NewRelic, following code:
```ruby
Curl::Easy.new.method(:to_s)
```
raises `` `method_with_tracing': wrong number of arguments (given 0, expected 1) (ArgumentError)``:

## Expected Behavior
An object like `#<Method: Curl::Easy(Kernel)#to_s()>` should be returned

## Steps to Reproduce
Paste following code into `newrelic_test.rb` and invoke `ruby newrelic_test.rb`

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem 'curb', '= 1.0.0'
  gem 'newrelic_rpm', '= 8.5.0'
end

Curl::Easy.new.method(:to_s)
```

## Your Environment
`ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]`

## Additional context
I believe the problem can be solved by changing [this line](https://github.com/newrelic/newrelic-ruby-agent/blob/8e453822c38b7fef829818883e81ae07ecc4a498/lib/new_relic/agent/instrumentation/curb/prepend.rb#L37) to
```ruby
method_with_tracing(verb) { super }
```

**Comment by fallwith** [comment]
Hello @knarewski,

Thanks very much for bringing this to our attention. Your explanation of the issue, steps to reproduce, and suggestion for a fix are all terrific and very much appreciated!

I believe you are spot on with your recommended fix. Via the `instrumentation.curb` setting inside of `newrelic.yml`, Curb can be instrumented via either `prepend` or `chain`. The chain implementation (`curb/chain.rb`) appears to correctly pass the `verb` variable over to Curb, but the prepend implementation does not. I believe this is an oversight/typo on our part and that we should be using `verb` in the manner that you suggest.

Would you be interesting in submitting a GitHub Pull Request against the `dev` branch of this repository with your recommended fix? The only requirement for doing so would be to sign our Contributor License Agreement when prompted to on the GitHub PR page after submitting. See the [Contributor License Agreement](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/CONTRIBUTING.md#contributor-license-agreement) section of our [contributing guide](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/CONTRIBUTING.md) for more details.

If you would prefer instead for the repo maintainers to make this change, we will be quite happy to do so.

**Comment by knarewski** [comment]
Hi @fallwith , thank you for a swift response and for confirming my suspicions! I'd prefer the repo maintainers to make the change if possible, thank you.

**Comment by fallwith** [comment]
Sure thing, @knarewski, the maintainers will work on the change and keep you posted.

**Comment by fallwith** [comment]
@knarewski's fix for this bug (thank you!) was merged with #1036 and should become available with the upcoming v8.6.0 release (ETA next 1-3 weeks).

If anyone wishes to leverage the fix before then, they are welcome to use the `curb_pass_verb_argument_to_method_with_tracing` branch of this repository.
